pinot-controller/src/test/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancerClusterStatelessTest.java

```diff
@@ -266,7 +311,7 @@ public void testRebalance()
     rebalanceConfig = new BaseConfiguration();
     rebalanceConfig.addProperty(RebalanceConfigConstants.REASSIGN_INSTANCES, true);
     rebalanceResult = tableRebalancer.rebalance(tableConfig, rebalanceConfig);
-    assertEquals(rebalanceResult.getStatus(), RebalanceResult.Status.DONE);
+    assertEquals(rebalanceResult.getStatus(), RebalanceResult.Status.NO_OP);
```

Comment:
I'm not sure if this is desired, but it's now possible for rebalance API to return no-op and the instance partition map to get created and/or deleted when changing from the default instance partition map. https://github.com/apache/pinot/blob/f7c5511ca3cb4478287059285529f516eebc8897/pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java#L418 I think the partition map should still be the same though, it's just that the ZK entry will not exist.